### DISCUSSION TITLE
Versions after 1.3 automatically reload at a much slower speed than 1.2.24

### DISCUSSION THREAD

**Discussion by lqblu** [zauberzeug/nicegui#1589] [Q&A] [answered]
### Question

I did a comparison in the same Python environment, and found that the automatic reload of versions after 1.3 is much slower than that of 1.2.24, sometimes keep showing: WARNING:  WatchFiles detected changes in 'xxxx.py'. Reloading... , but the web page is not updated. why?

**Comment by rodja** [comment]
>  sometimes keep showing: WARNING: WatchFiles detected changes in 'xxxx.py'. Reloading... , but the web page is not updated

Do you experience that with a simple hello-world example or only in your codebase?

> I did a comparison in the same Python environment, and found that the automatic reload of versions after 1.3 is much slower than that of 1.2.24

Strange. I'll need to perform my own tests but have not felt any delays while using NiceGUI reloads while working with it regularly. About how "slow" are we talking? More than 10 seconds?

**Reply by lqblu** [reply]
> Do you experience that with a simple hello-world example or only in your codebase?

I experience that with both(hello-world example and my code).

> About how "slow" are we talking? More than 10 seconds?

More than 10 seconds, and sometimes no response at all

**Reply by rodja** [reply]
> More than 10 seconds, and sometimes no response at all

Thats waaaay to long. Does it happen in all browsers? Which OS are you using? Which Python Version? Are there any errors in the javascript console?

**Reply by lqblu** [reply]
I only used edge browser on windows11, python version 3.8.0
![1694569112166](https://github.com/zauberzeug/nicegui/assets/114222511/f3e83b81-269d-4ed3-9cd7-dd50a35d9968)

**Reply by rodja** [reply]
Can you test on Chrome and Firefox? Also it would be interesting to see the network traffic when reloading...

**Reply by lqblu** [reply]
First of all, I would like to apologize for replying so late, because my access to Github is too slow. I used Firefox and the problem was the same.

**Reply by rodja** [reply]
Thanks. Are you starting the app via Terminal or PyCharm or another IDE? There is an [open PyCharm bug](https://youtrack.jetbrains.com/issue/PY-60962/os.killpid-signal.CRTLCEVENT-ignored-when-running-python-program-from-Pycharm) about slowness of reloading with uvicorn (which is used in NiceGUI): https://github.com/encode/uvicorn/issues/2000.

**Reply by lqblu** [reply]
The IDE I using is PyCharm, so is there no solution for this PyCharm bug?

**Reply by falkoschindler** [reply, marked as answer]
Then it's certainly caused by the bug @rodja mentioned:
https://youtrack.jetbrains.com/issue/PY-60962/os.killpid-signal.CRTLCEVENT-ignored-when-running-python-program-from-Pycharm
For a workaround or solution you might want to ask over there, on StackOverflow or a different PyCharm-related forum, since we normally don't use PyCharm and there's nothing we can do about it from NiceGUI's perspective.

**Comment by akai-1024** [comment]
I have also encountered a similar situation, which takes a long time to restart in the pycharm running state.
But it would be slightly better with the command-line tools.

**Reply by rodja** [reply]
> But it would be slightly better with the command-line tools.

What do you mean with "slightly better"? Is a simple "hello world" app loading in about 3 seconds?

**Reply by akai-1024** [reply]
I use command-line tools now, it reloading the speed is fast. Thanks, mate.
